### PR TITLE
Cr 423 rabbit support for cucumber tests

### PR DESCRIPTION
**Pull request by PeterBochel**
Several changes have been made to support cucumber tests which use Rabbit.

- Moved NativeRabbitEventSender to common-service.
- Added Rabbit endpoint. For use by either manual testing or shell-script tests.
- Moved timeout parsing (eg, 250ms, 2.5s) to it's own class, so that it can be shared between Firestore and Rabbit endpoints.

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/integration/event/generator/NativeRabbitEventGeneratorExample.java [hunk 1]
```diff
@@ -19,7 +21,7 @@ public static void main(String[] args) {
     try {
       connectionDetails = new RabbitConnectionDetails();
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/integration/event/generator/NativeRabbitEventGeneratorExample.java]: ?

File: src/main/java/uk/gov/ons/ctp/integration/event/generator/endpoint/RabbitEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  /**
+   * Note that this endpoint is only provided for manual testing of the RabbitHelper.sendEvent(). It
+   * is not intended to be used.
+   *
+   * <p>It basically duplicates the whole point of the EventGenerator, and only works with limited
+   * fixed data.
+   */
+  @RequestMapping(value = "/rabbit/send", method = RequestMethod.GET)
+  @ResponseStatus(value = HttpStatus.OK)
+  public ResponseEntity<String> sendEvent() throws Exception {
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/integration/event/generator/endpoint/RabbitEndpoint.java]: remove

**Review by philwhiles — CHANGES_REQUESTED** [review]
as per comments

File: src/main/java/uk/gov/ons/ctp/integration/event/generator/NativeRabbitEventGeneratorExample.java (hunk 1 above)
Comment (PeterBochel) [reply src/main/java/uk/gov/ons/ctp/integration/event/generator/NativeRabbitEventGeneratorExample.java]: I think that this is ok where it is. Its naming makes it look like it's in the wrong place.

It makes use of the EventGenerator so really has to live in this project. It's also using the NativeRabbitEventSender hence the use of the connection details object.

File: src/main/java/uk/gov/ons/ctp/integration/event/generator/endpoint/RabbitEndpoint.java (hunk 2 above)
Comment (PeterBochel) [reply src/main/java/uk/gov/ons/ctp/integration/event/generator/endpoint/RabbitEndpoint.java]: Method deleted.

File: src/main/java/uk/gov/ons/ctp/integration/event/generator/NativeRabbitEventGeneratorExample.java (hunk 1 above)
Comment (PeterBochel) [reply src/main/java/uk/gov/ons/ctp/integration/event/generator/NativeRabbitEventGeneratorExample.java]: After discussion, I'm deleting this class.

**Pull request merged**